drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -120,6 +125,18 @@ static int udc_rpi_start_xfer(uint8_t ep, const void *data, size_t len)
 static void udc_rpi_handle_setup(void)
 {
 	struct cb_msg msg;
+	struct udc_rpi_ep_state *ep_state = udc_rpi_get_ep_state(USB_CONTROL_EP_OUT);
+
+	if (*ep_state->buf_ctl & USB_BUF_CTRL_AVAIL) {
+		/* Un-arm EP0_OUT endpoint, to make sure next setup packet starts clean */
```

Comment:
DONE. Note that as I was writing the comment I realized that the more detailed comment didn't really make sense without also including a change that I was planning for a following PR, so I have included that here as well. I have also updated the commit message with the following lines added to the end:
     - When a setup req interrupt fires, don't immediately force the next
       read to get it, instead, it will be read only after a setup message
       is extracted from the message queue.
     - When a setup packet is received abort any unfinished previous control
       transfers:
        - cancel any data buffers given to the EP0_IN endpoint
        - drop any new ep0_in writes that are attempted before this newest
          setup packet's associated message is extracted from the message
          queue.

Reply:
So please see the new beginning of the udc_rpi_handle_setup() call and see if it makes sense.